examples/api-deploy-gke-nebius-okta/README.md

```diff
@@ -0,0 +1,296 @@
+
+# Temp variables used in the guide, no need to change
+TMP_KUBECONFIG=/tmp/sky_kubeconfig
+```
+
+### Prepare GCP credentials
+Create a secret with the GCP service account json key:
+
+```bash
+rm -f $TMP_KUBECONFIG # Remove the file if it exists
+
+# Get GKE credentials
+KUBECONFIG=$TMP_KUBECONFIG gcloud container clusters get-credentials $GKE_CLUSTER_NAME --zone $GKE_ZONE
+GKE_CONTEXT=$(kubectl config current-context --kubeconfig $TMP_KUBECONFIG)
+
+kubectl create namespace $NAMESPACE --kubeconfig $TMP_KUBECONFIG
+kubectl create secret generic gcp-credentials --kubeconfig $TMP_KUBECONFIG --context $GKE_CONTEXT \
+  --namespace $NAMESPACE \
+  --from-file=gcp-cred.json=$GCP_SERVICE_ACCOUNT_JSON
+```
+
+### Prepare Nebius credentials
+
+Create a secret with the Nebius service account json key:
+
+```bash
+# Set up Nebius credentials for Nebius CLI auth
+kubectl create secret generic nebius-credentials \
+  --namespace $NAMESPACE \
+  --from-file=credentials.json=$NEBIUS_SERVICE_ACCOUNT_JSON
+```
+
+### Prepare Kubernetes credentials: Nebius Managed Kubernetes and GKE
+
+Combine Nebius and GKE credentials into a single kubeconfig and create a secret with the combined kubeconfig:
+```bash
+# Get Nebius credentials
+nebius mk8s cluster get-credentials --id $NEBIUS_CLUSTER_ID --external --kubeconfig $TMP_KUBECONFIG
+NEBIUS_CONTEXT=$(kubectl config current-context --kubeconfig $TMP_KUBECONFIG)
+
+# To help SkyPilot identify GPUs on the Nebius cluster, label the nodes with the GPU type. If you already installed the Nvidia Device Plugin, this step will be automatically skipped.
+KUBECONFIG=$TMP_KUBECONFIG python -m sky.utils.kubernetes.gpu_labeler --context $NEBIUS_CONTEXT
+
+# Get GKE credentials
+KUBECONFIG=$TMP_KUBECONFIG gcloud container clusters get-credentials $GKE_CLUSTER_NAME --zone $GKE_ZONE
+
+# Verify both contexts are available in the kubeconfig
+kubectl config get-contexts --kubeconfig $TMP_KUBECONFIG
+
+# Strip exec paths from the kubeconfig to avoid hardcoded paths in the kubeconfig
+python -m sky.utils.kubernetes.exec_kubeconfig_converter --input $TMP_KUBECONFIG --output kubeconfig.converted
+
+# Create a secret with the converted kubeconfig
+kubectl create secret generic kube-credentials --kubeconfig $TMP_KUBECONFIG --context $GKE_CONTEXT \
+  --namespace $NAMESPACE \
+  --from-file=config=kubeconfig.converted
+
+# Create a SkyPilot config that allows both contexts (GKE and Nebius) to be used simultaneously:
+cat <<EOF > config.yaml
+kubernetes:
+  allowed_contexts:
+    - $NEBIUS_CONTEXT
+    - $GKE_CONTEXT
+EOF
+CONFIG_PATH=$PWD/config.yaml
+```
+
+
+## Step 2: Deploy the API server
+
+Deploy the API server with helm:
+
+```bash
+helm repo add skypilot https://helm.skypilot.co
+helm repo update
+
+helm upgrade --install skypilot skypilot/skypilot-nightly --devel \
+  --namespace $NAMESPACE \
+  --create-namespace \
+  --set-file apiService.config=$CONFIG_PATH \
```

Comment:
cc'ing @aylei as discussed offline, this may cause the config immutable from the dashboard with your proposed solution.

Reply:
Maybe I'm misunderstanding - Is this about some other proposal? As of this branch, the config can be edited and saved from the dashboard